flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -666,19 +666,17 @@ struct BoxRankOpConversion : public FIROpConversion<fir::BoxRankOp> {
   }
 };
 
+/// Lower `fir.boxproc_host` operation. Extracts the host pointer from the
+/// boxproc.
+/// TODO: Part of supporting Fortran 2003 procedure pointers.
 struct BoxProcHostOpConversion : public FIROpConversion<fir::BoxProcHostOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::BoxProcHostOp boxprochost, OperandTy operands,
+  matchAndRewrite(fir::BoxProcHostOp boxprochost, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    auto a = operands[0];
-    auto ty = convertType(boxprochost.getType());
-    auto c1 = mlir::ArrayAttr::get(boxprochost.getContext(),
-                                   rewriter.getI32IntegerAttr(1));
-    rewriter.replaceOpWithNewOp<mlir::LLVM::ExtractValueOp>(boxprochost, ty, a,
-                                                            c1);
-    return success();
+    TODO(boxprochost.getLoc(), "fir.boxproc_host codegen");
+    return failure();
```

Comment:
I think this one can "diverge" between fir-dev and upstream. The pattern looks plausible and serves as documentation if it there are bugs, etc.

Reply:
Done.